### DISCUSSION TITLE
[Web] Have the Escape key clear the search

### DISCUSSION THREAD

**Discussion by bjherbison** [ente-io/ente#11006] [Enhancements]
**Is your feature request related to a problem? Please describe.**
In many interfaces pressing Escape clears something recently set. In web browsers and many other applications Escape clears a search string. Since that is a common function of Escape I'm frustrated when Escape handling isn't implemented by an application or web page.

**Describe the solution you'd like**
When the Escape (Esc) is pressed on https://auth.ente.io/auth clear the search string and set focus to the search field.

**Comment by lokicodess** [comment]
I would love to work on this issue . Can you please assign me this issue.

**Comment by aakankshabhende** [comment]
@mnvr I would like to work on this issue. Could you please assign me this?

Is this issue intended for below search bar?

<img width="1717" alt="image" src="https://github.com/ente-io/ente/assets/63249668/fdc58c03-c2bb-4c58-a2ae-12cc28bc7d74">

**Comment by bjherbison** [comment]
> @mnvr I would like to work on this issue. Could you please assign me this?
> 
> Is this issue intended for below search bar?
> <img alt="image" width="1717" src="https://private-user-images.githubusercontent.com/63249668/310568967-fdc58c03-c2bb-4c58-a2ae-12cc28bc7d74.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3NDMyOTcsIm5iZiI6MTcwOTc0Mjk5NywicGF0aCI6Ii82MzI0OTY2OC8zMTA1Njg5NjctZmRjNThjMDMtYzJiYi00YzU4LWEyYWUtMTJjYzI4YmM3ZDc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDE2MzYzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljOGZkMWZmNjExOGE4MDg5YjZmMGVlNDNjM2FiYmNlNjAyNGM3MTM3NjI1NDc5NzQ5MjUxMzY5MTQ2ZDE0ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.n1KYXY_lvtzEiV8O8riAt82EIIYFUOrvi1qVwsLfgas">

I was intending the comment for https://auth.ente.io/auth but I didn't really specify that. It would apply for any search bar in any web application.

**Comment by mnvr** [comment]
Hey @aakankshabhende, thank you for helping out! As @bjherbison mentioned, this behaviour perhaps makes sense for most (all?) search bars, so we can add it to both auth and photos search bars. But if it is simpler for you then we can start by adding it to auth first, just to reduce the scope of changes.

**Comment by klaas0** [comment]
It would also be cool to focus the search bar with key combination (CTRL + F)

**Comment by daeh** [comment]
Adding support for escape to exit search in MacOS Desktop app (so that it's easy to see the tags). And for cmd-F to activate the search.

**Comment by SlimJim192** [comment]
> It would also be cool to focus the search bar with key combination (CTRL + F)

Yep - perhaps even just default to search if window focused and you start typing.. What else could you be trying to do.

I just want to open the app, type what I want, and only use the mouse to grab the code and leave. No back and forth.

**Comment by klaas0** [comment]
> > It would also be cool to focus the search bar with key combination (CTRL + F)
> 
> Yep - perhaps even just default to search if window focused and you start typing.. What else could you be trying to do.
> 
> I just want to open the app, type what I want, and only use the mouse to grab the code and leave. No back and forth.

Well there is actually an option for that. Settings -> General -> Focus search on app start
![image](https://github.com/user-attachments/assets/dc0430d3-078b-4ced-8435-d055ea0d4a79)

**Comment by bjherbison** [comment]
I don't see any Settings option on the web page.

Focus on the web page starts in the search, but once you click to get a code focus is off the text box. That's why I opened this issue and #478 .

**Comment by klaas0** [comment]
> I don't see any Settings option on the web page.
> 
> Focus on the web page starts in the search, but once you click to get a code focus is off the text box. That's why I opened this issue and #478 .

I see, well the setting are only available in the app. You can download the app for windows with this [link](https://github.com/ente-io/ente/releases/download/auth-v4.0.0/ente-auth-v4.0.0-installer.exe). Other installers are available on their [website](https://ente.io/auth/)

I would highly recommend the app, it feels very snappy!

**Comment by bjherbison** [comment]
I gave up Windows over a decade ago. The link for Linux download goes to a page with [no download option](https://github.com/ente-io/ente/releases?q=tag%3Aauth-v4). And once you figure out what to click after that (many links are underlined in the traditional web style, but what you need to click isn't) [the installer doesn't work](https://github.com/ente-io/ente/issues/2531). Once that is fixed I might try it, but I like it in a web page.
